improver_tests/nbhood/square_kernel/test_SquareNeighbourhood.py

```diff
@@ -209,42 +204,13 @@ def test_masked_array_re_mask_false(self):
             ]
         )
         self.cube.data = np.ma.masked_where(mask == 0, self.cube.data)
-        result = Neighbourhood("square", re_mask=False).run(self.cube, self.RADIUS)
-        self.assertArrayAlmostEqual(result.data, expected_array)
-
-    def test_nan_array_re_mask_true(self):
```

Comment:
No longer support unmasked NaNs. I couldn't think of a reason why we'd ever need to, so unmasked NaNs are now caught by the base class. No longer need tests here